ios/browser/api/bookmarks/brave_bookmarks_api.mm

```diff
@@ -426,9 +426,9 @@ - (void)moveToParent:(IOSBookmarkNode*)parent index:(NSUInteger)index {
   }
 }
 
-- (NSInteger)indexOfChild:(IOSBookmarkNode*)child {
+- (size_t)indexOfChild:(IOSBookmarkNode*)child {
   DCHECK(node_);
-  return node_->GetIndexOf(child->node_);
+  return *node_->GetIndexOf(child->node_);
```

Comment:
Two things here:

1. `GetIndexOf` returns an optional now, so the Obj-C version needs to check that and return `NSNotFound` here when its `nullopt`. Previously it would return `-1` instead of null, which matches `NSNotFound` so no changes would be needed on brave-ios side
2. We usually stick to Obj-C types for headers/wrapped code, and along with using `NSNotFound` we should keep the Obj-C return type as `NSInteger` and `static_cast` from `size_t` when not null.

Reply:
@cdesouza-chromium Can maybe take advantage of `absl::value_or()` to return `NSNotFound` if optional is empty?

Reply:
Fixed in c35fbf21313a880d33151c64557e9d0b27bebac3.